src/gurt/telemetry.c

```diff
@@ -1888,14 +2112,25 @@ d_tm_get_gauge(uint64_t *val, uint64_t *shmem_root, struct d_tm_node_t *node,
 	if (!d_tm_validate_shmem_ptr(shmem_root, (void *)node))
 		return -DER_METRIC_NOT_FOUND;
 
-	if (node->dtn_type != D_TM_GAUGE)
+	if (!(node->dtn_type & D_TM_DURATION))
```

Comment:
Just noticed the difference here between detecting type duration (treated as a single bit in a bitmap) and type gauge (treated as an enum, expected no other bits flipped). Intentional?

Reply:
It is intentional.  The D_TM_GAUGE type does not accept any modifiers.  The D_TM_DURATION and D_TM_TIMER_SNAPSHOT allow the caller to specify D_TM_CLOCK_REALTIME, D_TM_CLOCK_PROCESS_CPUTIME or D_TM_CLOCK_THREAD_CPUTIME to describe the timer type.  An improvement I have in mind (for a future PR) is to flatten that out so that rather than use D_TM_DURATION and D_TM_TIMER_SNAPSHOT, we use something that incorporates the timer type in the name so that only a single value is used to define the metric.  For example, we might have D_TM_DURATION_REALTIME, D_TM_DURATION_PROCESS_CPUTIME, D_TM_DURATION_THREAD_CPUTIME, D_TM_SNAPSHOT_REALTIME, D_TM_SNAPSHOT_PROCESS_CPUTIME, and D_TM_SNAPSHOT_THREAD_CPUTIME.  That would simplify things a bit.

Reply:
Understood. Thanks for the clarification!